operator-common/src/main/java/io/strimzi/operator/common/operator/resource/AbstractNamespacedResourceOperator.java

```diff
@@ -262,6 +303,23 @@ protected T patchOrReplace(String namespace, String name, T desired)   {
         return operation().inNamespace(namespace).withName(name).patch(PatchContext.of(PatchType.JSON), desired);
     }
 
+    protected T patch(Reconciliation reconciliation, String namespace, String name, T desired) {
```

Comment:
What is the reasoning behind having `patch` and `internalPatch`? I'm not sure I really understand the need fo the two methods. Maybe this is where the Javadocs would help.